src/Razor/Microsoft.NET.Sdk.Razor/src/DotnetToolTask.cs

```diff
@@ -40,7 +40,7 @@ public abstract class DotNetToolTask : ToolTask
 
         public string PipeName { get; set; }
 
-        protected override string ToolName => "dotnet";
+        protected override string ToolName => Path.GetDirectoryName(DotNetPath);
```

Comment:
Is this right?  This is now going to return a directory name?

Reply:
Come to think of it, it does seem incorrect and does not match the description of the property. I copied this pretty much verbatim from other places that do this, so maybe it's just one bad copy that's spread:

* https://github.com/mono/linker/blob/6252b94f0a9089cd717524be91879bb16a88937b/src/ILLink.Tasks/LinkTask.cs#L225
* https://github.com/dotnet/aspnetcore/blob/cd61ed769ab5fadcd7164fd395178442f1079b8d/src/Components/WebAssembly/Sdk/src/BrotliCompress.cs#L36

Reply:
Do we want to follow up on those as well?

I'll approve this for now, since it seems like it will work at least.

Reply:
Done - https://github.com/dotnet/aspnetcore/issues/25795